packages/styled-components/src/models/StyledComponent.js

```diff
@@ -57,7 +57,10 @@ function useResolvedAttrs<Config>(theme: any = EMPTY_OBJECT, props: Config, attr
 
     /* eslint-disable guard-for-in */
     for (key in resolvedAttrDef) {
-      context[key] = resolvedAttrs[key] = resolvedAttrDef[key];
+      context[key] = resolvedAttrs[key] =
+        key === 'className'
+          ? [resolvedAttrs[key], resolvedAttrDef[key]].filter(Boolean).join(' ')
```

Comment:
This may be a matter of taste, but we could just write a small `joinNames` function, which is then probably inline by Terser and saves us the intermediary array, e.g

```js
const joinNames = (a, b) =>
  a && b ? `${a} ${b}` : (a || b || “”)